### PR TITLE
Micromamba: Substitute env vars in .condarc

### PR DESCRIPTION
**Pull request by jonashaag**
New version of https://github.com/mamba-org/mamba/pull/1416 with only `${var}` support (and `$$` support)

### CODE REVIEW

**Comment by jonashaag** [comment]
@wolfv this should be ready to go with one minor todo where my C++ skills weren't good enough. How do you make a function that can return a `std::string` or "None"? I guess you make it return a pointer, but then where do you allocate the `std::string`? Does `std::string x` allocate the string on the stack or on the heap? If on the heap, how is it deallocated? What's the "modern" approach to heap allocation in C++?

**Comment by wolfv** [comment]
@jonashaag there are two options -- either to use `std::optional<std::string>` or `std::unique_ptr<std::string>`. Both will destroy the string when they go out of scope.

**Comment by wolfv** [comment]
@jonashaag can you have a look to see if this is what you had in mind?

Now when the env var exists, but is empty, it will add an empty string (`""`).

**Comment by jonashaag** [comment]
@wolfv I was about to push my changes at this very moment :D

**Comment by wolfv** [comment]
Nooooooooo! Sorry :/

**Comment by jonashaag** [comment]
https://github.com/jonashaag/mamba/tree/envsubst2-2

Can you review this instead?

File: libmamba/src/api/configuration.cpp [hunk 1]
```diff
@@ -51,8 +51,17 @@ namespace mamba
...
             else
             {
-                // TODO: change this so that empty but existing env vars are replaced with ""
-                val = env::get(var);
+                auto env_var = env::get(var);
+                if (env_var)
+                {
+                    val = env_var.value();
+                    if (val.empty())
+                        val = "\"\"";
```
Comment (jonashaag) [inline libmamba/src/api/configuration.cpp:59]: Sorry wasn't clear here, should be replaced by empty string

**Comment by jonashaag** [comment]
But I'm happy to see that you made essentially the same changes :) Are you ok with the change in behaviour that empty env vars are now not considered absent anymore? (You also seem to have made that decision in your changes)

**Comment by wolfv** [comment]
I am happy with your version, too. Do you want to force push to this branch to erase my changes and use yours?
I just read them and it's really super similar!

Sorry again, I should've coordinated :)

**Comment by jonashaag** [comment]
No worries. Force pushed.

**Comment by jonashaag** [comment]
@wolfv I don't understand the remaining Windows test failure. It seems that Micromamba thinks that `${foo}` is missing (not only empty!) and thus doesn't replace it. But the `env::get` code should be taking proper care of empty vs missing.

**Comment by wolfv** [comment]
I think 
it's a Windows limitation. Setting empty strings to environment variables seems unsupported

So IMHO we can skip that test on Windows.

Screenshot 2022-02-01 at 21 43 12

**Comment by jonashaag** [comment]
Oh wow, well that's unfortunate but then we can simplify some of the code and skip that test.

File: micromamba/tests/test_config.py [hunk 2]
```diff
@@ -14,19 +16,21 @@ def config(*args):
     cmd = [umamba, "config"] + [arg for arg in args if arg]
     res = subprocess.check_output(cmd)
     if "--json" in args:
-        j = json.loads(res)
+        j = yaml.load(res, yaml.FullLoader)
```
Comment (wolfv) [inline micromamba/tests/test_config.py:19]: I think you've filed a bug for this already, right?

Comment (jonashaag) [reply micromamba/tests/test_config.py:19]: Yes

**Comment by wolfv** [comment]
Thanks for tackling this!

**Pull request merged**